Refactoring of `StudyDirection`. (#1090)
optuna/optuna: docs/source/reference/structs.rst

.. autoclass:: StudyDirection
    :members:

>     .. deprecated:: 1.3.0

Reviewer: This change will first be included in the next release.
```suggestion
    .. deprecated:: 1.4.0
```

Author: Thank you for your comment. I will change it as you pointed.

>             The use of this class is deprecated and it is recommended that you use :class:`~optuna.study.StudyDirection` instead.

Reviewer 2: [nit-picking] Let's simplify this.
```suggestion
            This class is deprecated. Please use :class:`~optuna.study.StudyDirection` instead.
```
